api/externalLinks/ExternalLinksRestHandler.go

```diff
@@ -0,0 +1,218 @@
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package externalLinks
+
+import (
+	"encoding/json"
+	"github.com/devtron-labs/devtron/api/restHandler/common"
+	delete2 "github.com/devtron-labs/devtron/pkg/delete"
+	"github.com/devtron-labs/devtron/pkg/externalLinks"
+	"github.com/devtron-labs/devtron/pkg/user"
+	"github.com/devtron-labs/devtron/pkg/user/casbin"
+	"github.com/gorilla/mux"
+	"github.com/juju/errors"
+	"go.uber.org/zap"
+	"gopkg.in/go-playground/validator.v9"
+	"net/http"
+	"strconv"
+)
+
+type ExternalLinksRestHandler interface {
+	CreateExternalLinks(w http.ResponseWriter, r *http.Request)
+	GetExternalLinksTools(w http.ResponseWriter, r *http.Request)
+	GetExternalLinks(w http.ResponseWriter, r *http.Request)
+	UpdateExternalLinks(w http.ResponseWriter, r *http.Request)
+	DeleteExternalLinks(w http.ResponseWriter, r *http.Request) // Update is_active to false link
+}
+type ExternalLinksRestHandlerImpl struct {
+	logger               *zap.SugaredLogger
+	externalLinksService externalLinks.ExternalLinksService
+	userService          user.UserService
+	validator            *validator.Validate
+	enforcer             casbin.Enforcer
+	userAuthService      user.UserAuthService
+	deleteService        delete2.DeleteService
+}
+
+func NewExternalLinksRestHandlerImpl(logger *zap.SugaredLogger,
+	externalLinksService externalLinks.ExternalLinksService,
+	userService user.UserService,
+	enforcer casbin.Enforcer,
+	validator *validator.Validate, userAuthService user.UserAuthService,
+	deleteService delete2.DeleteService,
+) *ExternalLinksRestHandlerImpl {
+	return &ExternalLinksRestHandlerImpl{
+		logger:               logger,
+		externalLinksService: externalLinksService,
+		userService:          userService,
+		validator:            validator,
+		enforcer:             enforcer,
+		userAuthService:      userAuthService,
+		deleteService:        deleteService,
+	}
+}
+func (impl ExternalLinksRestHandlerImpl) CreateExternalLinks(w http.ResponseWriter, r *http.Request) {
+	userId, err := impl.userService.GetLoggedInUser(r)
+	if userId == 0 || err != nil {
+		common.WriteJsonResp(w, err, "Unauthorized User", http.StatusUnauthorized)
+		return
+	}
+	decoder := json.NewDecoder(r.Body)
+	var beans []*externalLinks.ExternalLinksRequest
+	err = decoder.Decode(&beans)
+	if err != nil {
+		impl.logger.Errorw("request err, SaveLink", "err", err, "payload", beans)
+		common.WriteJsonResp(w, err, nil, http.StatusBadRequest)
+		return
+	}
+
+	token := r.Header.Get("token")
+	if ok := impl.enforcer.Enforce(token, casbin.ResourceGlobal, casbin.ActionCreate, "*"); !ok {
```

Comment:
Is this the super-user check? Please confirm

Reply:
yes